Elevel/Application/Features/TestCommands/GetAllTestsQuery.cs

```diff
@@ -25,25 +27,35 @@ public class Request : PagedQueryBase, IRequest<Response>
 
             public Guid? UserId { get; set; }
 
+            [JsonIgnore]
+            public Guid SearchingUserId { get; set; }
+
             public Guid? Id { get; set; }
         }
 
         public class Handler : IRequestHandler<Request, Response>
         {
             private readonly IApplicationDbContext _context;
-
+            private readonly UserManager<User> _userManager;
             private readonly IMapper _mapper;
 
-            public Handler(IApplicationDbContext context, IMapper mapper)
+            public Handler(IApplicationDbContext context, IMapper mapper, UserManager<User> userManager)
             {
                 _context = context;
-
+                _userManager = userManager;
                 _mapper = mapper;
             }
 
             public async Task<Response> Handle(Request request, CancellationToken cancellationToken)
             {
                 var tests = _context.Tests.AsNoTracking();
+                var user = await _userManager.Users.FirstOrDefaultAsync(x => x.Id == request.SearchingUserId);
```

Comment:
no, it should be at controller method level from token. Imagine that your auth server is on another server and you have no access to DB